mantidproject/mantid
# Add basic models to ISIS Reflectometry Interface

~~~**Wait for #22784 and update `isis_reflectometry_alternative_table` before merging?**~~~

**Description of work.**
- Adds cut and paste functionality to JobTreeView.
- Tidies up HintingLineEdit and HintStrategy to use a `vector<Hint>` rather than a map. This makes exporting `HintStrategy` to python via SIP easier.
- Adds a method to `JobTreeView` to allow setting a column to use a `HintingLineEdit`.
- Adds `Row`, `Group` and `ReductionJob` models.
- Adds `BatchPresenter` and `BatchView` which replace the table in the Reflectometry Interface.
- Replaces `ReflTransferStrategy` mechanism with a method on the `Jobs` variant `mergeIntoJobs`.

This PR also breaks a number of features on the Reflectometry GUI, this functionality will be restored in the later PRs.

**To test:**
Open the Reflectometry Interface and experiment with transferring runs from the search results.
Experiment with the run filtering functionality. 
Ensure it is possible to edit the name and angle of a run while the filtering is applied.


Refs #22263 . 



---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
